src/gsy_e/gsy_e_core/simulation/time_manager.py

```diff
@@ -155,11 +155,10 @@ def handle_slowdown_and_realtime_scm(self, slot_no: int, slot_count: int,
             log.debug("Slot %s/%s: Sleep time of %s s was applied",
                       slot_no, slot_count, sleep_time_s)
 
-        self.slot_time_counter = time()
+        self.slot_time_counter = int(time())
```

Comment:
I do not really get the reason for this change. AFAIK, `time()` return the current time in unix seconds as float, right? Adding `int()` will just strip of the milliseconds. This is not the actual fix right? So I propose to revert this here in oder to als take into account the milliseconds (this difference could add up to some seconds when the simulation has progressed a bit).

Reply:
Since the smallest amount of time in gsy-e is one second, fractions of seconds can add up and eventually change the slot time after some slots are passed. Please notice how it can affect slot time in the description of the PR.

Reply:
I just wrote a long comment, but had to remove it again, as I only see now, why it works.
The actual fix is that you added line 175 here, in order to re-set the `self.slot_time_counter` in `calc_resume_slot_and_count_realtime`. Great! 
However, for this we actually have the `reset` method in the `SimulationTimeManagerScm`. When I implemented it, I forgot to call it in the `CoefficientSimulation`. So, I propose to remove this line from `calc_resume_slot_and_count_realtime` and actually call reset in the `_execute_simulation`: 
<img width="1603" alt="Screenshot 2022-10-10 at 11 25 12" src="https://user-images.githubusercontent.com/19326608/194836609-3bd43cee-0aaa-4676-878c-039ec02d0293.png">

Reply:
I still not agree on cutting off the fractions of seconds. IMO they are needed to really set the actual end of the timeslot. The indeed add up, and if you always omit a fraction of a seconds, the start of the slots will be off when you run the simulation for longer.

Reply:
IMO, fractions of second are only because of the time used for calculating methods of simulation time manager. I think in an ideal world, the amount time needed to calculate time should be zero. That's why I suggest to get rid of second fractions.

Reply:
These fraction can add up do large amounts (when you accumulate them a lot of times, like we did once in the "normal" simulation where we track each tick). 
Here it does not seem to get added up along the way (judging from the PR description) which I do not understand why now. I give up though. Please leave the `int()` in and we can remove it when we actually have long simulations running and see a slight difference. We probably will not..
Thanks for bering with me!

Reply:
Thanks a lot